### DISCUSSION TITLE
Image sequencing is stuck at 00000 despite "Add number to filename" setting is on and [seq] is used?

### DISCUSSION THREAD

**Discussion by YourFriendlyNeighborhoodMONKE** [vladmandic/sdnext#2405] [Q&A] [answered]
My image saving sequence is as follows:
![image](https://github.com/vladmandic/automatic/assets/115096590/5a649dad-e1f5-4ef0-8966-31a957c22848)

This setting _should_ produce a filename pattern in a folder:
00000-prompt words, more prompt words
00001-prompt words, more prompt words
00002-prompt words, more prompt words
etc.

I don't know what happened, but [seq] seems to be broken for me for unknown reason

Now it just keeps on being stuck at the first one:
00000-prompt words, more prompt words

The new file will **no longer jump into the next number, but instead _overwrites_ the first file**

I've tried to look through the settings, but cannot find anything to fix this
I have tried to restart the server several times

Any ideas?

Linked code that is probably related:
[code.txt](https://github.com/vladmandic/automatic/files/13169391/code.txt)

**Comment by brknsoul** [comment]
workaround for now: add `[datetime]` to your file name. This will give a datetime code that should make your files unique.
I personally use `[seq]-[datetime]-[seed]-[prompt]` results in `00000-20231025064020-2742964881-promptwords.png`

**Reply by YourFriendlyNeighborhoodMONKE** [reply]
I used [image_hash] as a first remedy and sorted by date, but this is better,

Nice tip!

**Reply by YourFriendlyNeighborhoodMONKE** [reply]
@brknsoul Vlad just fixed it! :)

**Comment by vladmandic** [comment, marked as answer]
issue is that sequence is determined before filename sanitize, but if filename contains invalid characters, sequence will not match.
i've pushed an update. if issue persists, please set env variable `SD_PATH_DEBUG=true` and post log file showing the probematic behavior.

btw, `[prompt]` is allowed for backward compatibility, but i strongly suggest to use `[prompt_words]` instead - its much cleaner.

**Comment by YourFriendlyNeighborhoodMONKE** [comment]
> issue is that sequence is determined before filename sanitize, but if filename contains invalid characters, sequence will not match. i've pushed an update. if issue persists, please set env variable `SD_PATH_DEBUG=true` and post log file showing the probematic behavior.
> 
> btw, `[prompt]` is allowed for backward compatibility, but i strongly suggest to use `[prompt_words]` instead - its much cleaner.

Pulled and it is fixed for me!

Changed to [prompt_words] as suggested

Thank you Vlad! 👍
